Dockerfile.dev

```diff
@@ -0,0 +1,23 @@
+FROM ruby:2.5.8
+
+RUN curl -sS https://dl.yarnpkg.com/debian/pubkey.gpg | apt-key add -
+RUN echo "deb https://dl.yarnpkg.com/debian/ stable main" | tee /etc/apt/sources.list.d/yarn.list
+RUN apt-get update -qq && apt-get remove cmdtest && apt-get install -y nodejs postgresql-client yarn
+RUN gem install bundler
+ENV RAILS_ENV development
+ENV DATABASE_HOST postgresql
+WORKDIR /app
+COPY Gemfile /app/Gemfile
```

Comment:
So I think the intention at least was that if any of these four files change, we need to trigger a new bundle/yarn and copy the app again. If we have them on separate lines, then I expect Docker to cache the files that _haven't_ changed. If we have all four on one line, we copy four files even if we only changed one... not a big deal and might not even be true (I'm too completely confident with Docker) but doesn't seem like a huge deal to keep separate

Reply:
@dmgarland yes if any of those files change it will trigger a breaking of the cache and will rebuild from there.
I have basically taken this from multiple Dockerfiles I have seen and used it on my own projects.

We could as a 1/2 way house split it into js and ruby focused, as possibly if you alter ruby with say rails/webpacker updates then it would also have an impact on the js, vs. where a js only change will most likely not impact the ruby files

i.e

```
COPY Gemfile* /app/
COPY package.json yarn.lock /app/
```

Reply:
LGTM